CONTRIBUTING.md

```diff
@@ -59,3 +61,112 @@ And make sure that it is executable:
+1. **Fix or implement something in an existing plugin:** In this case you have to fork only the corresponding plugin. You can still use the [main repository of CosmoScout VR](https://github.com/cosmoscout/cosmoscout-vr).
+1. **Fix or implement something in an existing plugin which requires changes in the core engine:** In this case you have to do both, 1 and 2.
+1. **Create a new plugin:** In this case you do not have to fork anything, except if your new plugin requires some changes to the core engine. Then you would go for option 1.
+
+## 1. Forking only the main repository
+
+This is pretty straight-forward. Just click the **Fork** button on the top right of this page. Then clone the forked repository, perform your changes, push to a feature branch and create a pull request to CosmoScout's develop branch.
+
+``` bash
+git clone git@github.com:<your user name>/cosmoscout-vr.git
+cd cosmoscout-vr
+git remote add upstream git@github.com:cosmoscout/cosmoscout-vr.git
+git checkout develop
+git submodule update --init
+git checkout -b feature/your-new-feature
+
+# ... do and commit your changes!
+
+git push origin feature/your-new-feature
+```
+
+When there were changes in CosmoScout's develop branch, you will need to merge those to your fork before creating a pull request:
+
+``` bash
+git fetch upstream
+git merge upstream/develop
+```
+
+Then you can create a pull request on GitHub to CosmoScout's develop branch.
+
+## 2. Forking only a plugin
+
+In this case, you have to clone CosmoScout's [main repository](https://github.com/cosmoscout/cosmoscout-vr) and fork the plugin you want to modify. Once all plugin submodules have been checked out, you can add your forked plugin as a second remote to the plugin submodule.
+
+``` bash
+git clone git@github.com:cosmoscout/cosmoscout-vr.git
+cd cosmoscout-vr
+git checkout develop
+git submodule update --init
+cd plugins/csp-whatever
+git remote add myfork git@github.com:<your user name>/csp-whatever.git 
+git checkout -b feature/your-new-feature
+
+# ... do and commit your changes!
+
+git push -u myfork feature/your-new-feature
+```
+
+When there were changes in the plugin's develop branch, you will need to merge those to your fork before creating a pull request:
+
+``` bash
+cd plugins/csp-whatever
+git fetch origin
+git merge origin/develop
+```
+
+Then you can create a pull request on GitHub to CosmoScout's develop branch.
+
+## 3. Forking the main repository and a plugin
+
+This basically combines both approaches above. First you have to clone the main repository and the plugin you want to modify. Then clone your forked repository and add a second remote to the plugin submodule pointing to your forked plugin repository.
```

Comment:
"First you have to clone the main repository" should be fork the main repository